context.go

```diff
@@ -76,7 +76,7 @@ func (context *Context) Run(addressesToData map[string]any, timeout time.Duratio
 	}
 	obj, err := encoder.Encode(addressesToData)
 	if err != nil {
-		return nil, nil, err
+		return res, err
```

Comment:
```suggestion
		return nil, err
```

Reply:
This means changing the return type to `*Result`. I don't think it's needed since `err` is what is used to know if `res` should be used. An argument could be made for copy saving since the result bounces around a couple functions, but the structure is small and will get shallow copied so it should be neglectable.